### PR TITLE
#6647: disable failing ci cpp tests and reenable cpp pipeline on CI

### PR DESCRIPTION
**Pull request by vtangTT**

### CODE REVIEW

File: .github/workflows/fast-dispatch-build-and-unit-tests.yaml [hunk 1]
```diff
@@ -33,6 +33,7 @@ jobs:
           {name: misc group 2, cmd: pytest $TT_METAL_HOME/tests/tt_eager/python_api_testing/unit_testing/misc/ -vvv --splits 3 --group 2 },
           {name: misc group 3, cmd: pytest $TT_METAL_HOME/tests/tt_eager/python_api_testing/unit_testing/misc/ -vvv --splits 3 --group 3 },
           {name: sweep, cmd: ./tests/scripts/run_python_sweep_tests.sh},
+          {name: C++, cmd: ./tests/scripts/run_cpp_unit_tests.sh},
```
Comment (TT-billteng) [inline .github/workflows/fast-dispatch-build-and-unit-tests.yaml:36]: you should add it to `cpp-post-commit.yaml` rather than add it here

Comment (vtangTT) [reply .github/workflows/fast-dispatch-build-and-unit-tests.yaml]: should be changed now

File: tests/tt_metal/tt_metal/unit_tests_common/dprint/test_invalid_print_core.cpp [hunk 2]
```diff
@@ -12,9 +12,11 @@
 //////////////////////////////////////////////////////////////////////////////////////////
 using namespace tt::tt_metal;
 
-TEST(DPrintErrorChecking, TestPrintInvalidCore) {
+TEST(DPrintErrorChecking, DISABLED_TestPrintInvalidCore) {
```
Comment (TT-billteng) [inline tests/tt_metal/tt_metal/unit_tests_common/dprint/test_invalid_print_core.cpp:15]: issue # for this test?

File: tests/tt_metal/tt_metal/unit_tests_common/dprint/test_invalid_print_core.cpp [hunk 3]
```diff
@@ -12,9 +12,11 @@
...
     // Set DPRINT enabled on a mix of invalid and valid cores. Previously this would hang during
     // device setup, but not the print server should simply ignore the invalid cores.
+    tt::ARCH arch = tt::get_arch_from_string(tt::test_utils::get_env_arch_name());
```
Comment (TT-billteng) [inline tests/tt_metal/tt_metal/unit_tests_common/dprint/test_invalid_print_core.cpp:18]: isn't this redundant if you use`DISABLED_` ?

File: tests/tt_metal/tt_metal/unit_tests_common/dprint/test_print_all_harts.cpp [hunk 4]
```diff
@@ -177,7 +177,9 @@ static void RunTest(DPrintFixture* fixture, Device* device) {
     );
 }
 
-TEST_F(DPrintFixture, TestPrintFromAllHarts) {
+TEST_F(DPrintFixture, DISABLED_TestPrintFromAllHarts) {
+    tt::ARCH arch = tt::get_arch_from_string(tt::test_utils::get_env_arch_name());
+    if (arch == tt::ARCH::WORMHOLE || arch == tt::ARCH::WORMHOLE_B0) GTEST_SKIP_("see issue #6659");
```
Comment (TT-billteng) [inline tests/tt_metal/tt_metal/unit_tests_common/dprint/test_print_all_harts.cpp:182]: redundant if you use `DISABLED_` ?

File: tests/tt_metal/tt_metal/unit_tests_common/dprint/test_print_hanging.cpp [hunk 5]
```diff
@@ -52,10 +52,12 @@ try {
     );
 }
 
-TEST_F(DPrintFixture, TestPrintHanging) {
+TEST_F(DPrintFixture, DISABLED_TestPrintHanging) {
     // Skip this test for slow dipatch for now. Due to how llrt currently sits below device, it's
     // tricky to check print server status from the finish loop for slow dispatch. Once issue #4363
     // is resolved, we should add a check for print server handing in slow dispatch as well.
+    tt::ARCH arch = tt::get_arch_from_string(tt::test_utils::get_env_arch_name());
+    if (arch == tt::ARCH::WORMHOLE || arch == tt::ARCH::WORMHOLE_B0) GTEST_SKIP_("see issue #6659");
```
Comment (TT-billteng) [inline tests/tt_metal/tt_metal/unit_tests_common/dprint/test_print_hanging.cpp:60]: redundant?

File: tests/tt_metal/tt_metal/unit_tests_common/basic/test_device_init.cpp [hunk 6]
```diff
@@ -62,7 +62,7 @@ bool load_all_blank_kernels(tt_metal::Device* device) {
 
 INSTANTIATE_TEST_SUITE_P(DeviceInit, DeviceParamFixture, ::testing::Values(1, tt::tt_metal::GetNumAvailableDevices()));
 
-TEST_P(DeviceParamFixture, DeviceInitializeAndTeardown) {
+TEST_P(DeviceParamFixture, DISABLED_DeviceInitializeAndTeardown) { // see issue #6659
```
Comment (TT-billteng) [inline tests/tt_metal/tt_metal/unit_tests_common/basic/test_device_init.cpp:65]: Noticed that this only fails on N300 according to issue
 If so, we shouldn't use `DISABLED_` and skip if device is N300

Comment (vtangTT) [reply tests/tt_metal/tt_metal/unit_tests_common/basic/test_device_init.cpp:65]: when i was iterating the test will try to run and hang if i try using the skip, so i just disabled it completely

**Comment by vtangTT** [comment]
seems to be stable now: 
https://github.com/tenstorrent-metal/tt-metal/actions/runs/8411469809
https://github.com/tenstorrent-metal/tt-metal/actions/runs/8407116329

@TT-billteng

**Review by TT-billteng — APPROVED** [review]

**Pull request merged**